packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClockSection.tsx

```diff
@@ -155,9 +156,10 @@ export const MultiSectionDigitalClockSection = React.forwardRef(
       const selectedItem = containerRef.current.querySelector<HTMLElement>(
         '[role="option"][aria-selected="true"]',
       );
-      if (!selectedItem) {
+      if (!selectedItem || previousSelected.current === selectedItem) {
```

Comment:
> Not sure it can actually happen, but if `selectedItem` is null, shouldn't we update `previousSelected` so that the next update compared it to `null` and not to the previous previous value ?

Indeed, I'll add this case, but in reality, it might never be triggered given the current functionality as we do not have a UI for resetting a value. 🤷 

> And by the way, how stable is a reference comparison on an HTML element ?
Isn't there a risk that the element could change it's reference when re-rendering ?

Seems pretty stable. React shouldn't re-render the DOM if the component didn't change and after testing, it seems to work correctly. 👍